src/bio/bio_wal.c

```diff
@@ -890,6 +891,12 @@ bio_wal_commit(struct bio_meta_context *mc, struct umem_wal_tx *tx, struct bio_d
 		return rc;
 	}
 
+	act_nr = umem_tx_act_nr(tx) + dc_arr.dca_nr;
```

Comment:
I think we should just avoid calling bio_wal_commit() in AD (just like what BMEM does) when umem_tx_act_nr() == 0. Zero umem_tx_act_nr() with non-NULL data_iod (non-zero dc_arr.dca_nr) is impossible.